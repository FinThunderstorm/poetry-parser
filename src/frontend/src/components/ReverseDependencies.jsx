import React from 'react'
import {
    Table,
    TableBody,
    TableCell,
    TableContainer,
    TableHead,
    TableRow,
    Typography,
} from '@mui/material'
import { Link } from 'react-router-dom'

/** ReverseDependencies converts reverseDependencies array to table
 */
const ReverseDependencies = ({ reverseDependencies }) => (
    <>
        <Typography variant="button" color="text.secondary">
            Reverse dependencies:
        </Typography>
        <TableContainer sx={{ height: '60vh' }}>
            <Table stickyHeader size="small">
                <TableHead>
                    <TableRow>
                        <TableCell>Name</TableCell>
                    </TableRow>
                </TableHead>
                <TableBody>
                    {reverseDependencies.map((reverseDependency) => (
                        <TableRow key={reverseDependency.name}>
                            {reverseDependency.installed ? (
                                <TableCell>
                                    <Link
                                        to={`/packages/${reverseDependency.name}`}
                                        title={`reverseDependency-${reverseDependency.name}`}
                                    >
                                        {reverseDependency.name}
                                    </Link>
                                </TableCell>
                            ) : (
                                <TableCell>{reverseDependency.name}</TableCell>
                            )}
                        </TableRow>
                    ))}
                </TableBody>
            </Table>
        </TableContainer>
    </>
)

export default ReverseDependencies
