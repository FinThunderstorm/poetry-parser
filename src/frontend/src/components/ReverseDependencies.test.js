/* eslint-disable react/prop-types */
import React from 'react'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect'
import { render, screen } from '@testing-library/react'
import { MemoryRouter, Route, Routes } from 'react-router-dom'

import ReverseDependencies from './ReverseDependencies'

describe('ReverseDependencies', () => {
    const ReverseDependenciesWrapper = ({ url, reverseDependencies }) => (
        <MemoryRouter initialEntries={[url]}>
            <Routes>
                <Route path="/">
                    <Route path="packages">
                        <Route
                            path=":name"
                            element={
                                <ReverseDependencies
                                    reverseDependencies={reverseDependencies}
                                />
                            }
                        />
                    </Route>
                </Route>
            </Routes>
        </MemoryRouter>
    )

    test('ReverseDependencies renderes correctly', () => {
        const reverseDependencies = [
            {
                name: 'pytest',
                installed: true,
            },
            {
                name: 'tottoroo',
                installed: true,
            },
        ]
        render(
            <ReverseDependenciesWrapper
                url="/packages/atomicwrites"
                reverseDependencies={reverseDependencies}
            />
        )

        const title = screen.getByText('Reverse dependencies:', {
            exact: false,
        })
        const name = screen.getByText('Name', {
            exact: false,
        })

        expect(title).toBeDefined()
        expect(name).toBeDefined()

        const pytest = screen.getByText('pytest', {
            exact: false,
        })
        const tottoroo = screen.getByText('tottoroo', {
            exact: false,
        })

        expect(pytest).toBeDefined()
        expect(tottoroo).toBeDefined()

        const tree = renderer
            .create(
                <ReverseDependenciesWrapper
                    url="/packages/atomicwrites"
                    reverseDependencies={reverseDependencies}
                />
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})
