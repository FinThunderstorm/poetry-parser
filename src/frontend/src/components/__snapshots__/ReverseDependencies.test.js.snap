// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReverseDependencies ReverseDependencies renderes correctly 1`] = `
Array [
  <span
    className="MuiTypography-root MuiTypography-button css-1jm68wq-MuiTypography-root"
  >
    Reverse dependencies:
  </span>,
  <div
    className="MuiTableContainer-root css-e2p4vx-MuiTableContainer-root"
  >
    <table
      className="MuiTable-root MuiTable-stickyHeader css-xn82ks-MuiTable-root"
      role={null}
    >
      <thead
        className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeSmall css-1h37f49-MuiTableCell-root"
            scope="col"
          >
            Name
          </th>
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
          role={null}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
          >
            <a
              href="/packages/pytest"
              onClick={[Function]}
              title="reverseDependency-pytest"
            >
              pytest
            </a>
          </td>
        </tr>
        <tr
          className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
          role={null}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
          >
            tottoroo
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
]
`;
